c7n/resources/apigw.py

```diff
@@ -309,6 +309,198 @@ def augment(self, resources):
         return results
 
 
+@resources.register('rest-vpclink')
+class RestAPI(query.QueryResourceManager):
+
+    class resource_type(object):
+        service = 'apigateway'
+        type = None
+        dimension = None
+        enum_spec = ('get_vpc_links', 'items', None)
+        id = 'id'
+        filter_name = None
+        name = 'name'
+
+
+@RestResource.filter_registry.register('rest-integration')
+class FilterRestIntegration(ValueFilter):
+    """Filter rest resources based on a key value for the rest method integration of the api
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: api-method-integrations-with-type-aws
+            resource: rest-resource
+            filters:
+              - type: rest-integration
+                key: type
+                value: AWS
+    """
+
+    schema = utils.type_schema(
+        'rest-integration',
+        method={'type': 'string', 'enum': [
+            'all', 'ANY', 'PUT', 'GET', "POST",
+            "DELETE", "OPTIONS", "HEAD", "PATCH"]},
+        rinherit=ValueFilter.schema)
+    permissions = ('apigateway:GET',)
+
+    def process(self, resources, event=None):
+        method_set = self.data.get('method', 'all')
+        # 10 req/s with burst to 40
+        client = utils.local_session(
+            self.manager.session_factory).client('apigateway')
+
+        # uniqueness constraint validity across apis?
+        resource_map = {r['id']: r for r in resources}
+
+        futures = {}
+        results = set()
+
+        with self.executor_factory(max_workers=2) as w:
+            tasks = []
+            for r in resources:
+                r_method_set = method_set
+                if method_set == 'all':
+                    r_method_set = r.get('resourceMethods', {}).keys()
+                for m in r_method_set:
+                    tasks.append((r, m))
+            for task_set in utils.chunks(tasks, 20):
+                futures[w.submit(
+                    self.process_task_set, client, task_set)] = task_set
+
+            for f in as_completed(futures):
+                task_set = futures[f]
+
+                if f.exception():
+                    self.manager.log.warning(
+                        "Error retrieving integrations on resources %s",
+                        ["%s:%s" % (r['restApiId'], r['path'])
+                         for r, mt in task_set])
+                    continue
+
+                for i in f.result():
+                    if self.match(i):
+                        results.add(i['resourceId'])
+                        # resource_map[i['resourceId']].setdefault(
+                        #     'c7n-matched-resource-method-integrations', {}).setdefault(
+                        #         ['resourceHttpMethod'], []).append(i)
+                        resource_map[i['resourceId']].setdefault(
+                            'c7n-matched-resource-method-integrations', []).append(i)
```

Comment:
we're trying to standardize on `c7n:` as the standard prefix

Reply:
also might be worth both shortening the name and making it a constant, versus trying to reference the string everywhere.

Reply:
Fixed and changed c7n- to c7n: 

Also I kept the longer name for the annotation since I felt it was a good descriptor and matched the parent child relationship correctly i.e. resource->method->integration. Let me know if you still want me to change the name but I think it works.